daemon/client.go

```diff
@@ -61,7 +75,17 @@ func (c *client) ExecuteProc(name string, args map[string]string) (string, error
 		return "", err
 	}
 
-	resp, err := http.Post("http://"+c.proctorEngineURL+"/jobs/execute", "application/json", bytes.NewReader(requestBody))
+	client := &http.Client{}
+	req, err := http.NewRequest("POST", "http://"+c.proctorEngineURL+"/jobs/execute", bytes.NewReader(requestBody))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Email-Id", c.emailId)
+	req.Header.Add("Access-Token", c.accessToken)
+	resp, err := client.Do(req)
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return "", errors.New(http.StatusText(http.StatusUnauthorized))
+	}
+
 	if err != nil || resp.StatusCode != http.StatusCreated {
 		return "", err
```

Comment:
When status code isn't 200, and error is actually nil, this condition becomes true.

So maybe break the condition like this:
```
if err != nil {
  return "", err
}

if resp.StatusCode != http.StatusCreated {
		return "", errors.New("error submitting job for execution")
}
```